Fix publish action

The publish action always fails. Looks like the failure is:

```
2022-04-13T21:54:31.3728878Z gpg: directory '/home/runner/.gnupg' created
2022-04-13T21:54:31.3731461Z gpg: keybox '/home/runner/.gnupg/pubring.kbx' created
2022-04-13T21:54:31.3745412Z gpg: no default secret key: No secret key
2022-04-13T21:54:31.3747776Z gpg: signing failed: No secret key
2022-04-13T21:54:31.3783977Z [INFO] ------------------------------------------------------------------------
2022-04-13T21:54:31.3784287Z [INFO] BUILD FAILURE
2022-04-13T21:54:31.3784830Z [INFO] ------------------------------------------------------------------------
2022-04-13T21:54:31.3798155Z [INFO] Total time:  16.112 s
2022-04-13T21:54:31.3839441Z [INFO] Finished at: 2022-04-13T21:54:31Z
2022-04-13T21:54:31.3839855Z [INFO] ------------------------------------------------------------------------
2022-04-13T21:54:31.3840625Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:1.6:sign (sign-artifacts) on project javarosa: Exit code: 2 -> [Help 1]
2022-04-13T21:54:31.3841071Z [ERROR] 
2022-04-13T21:54:31.3841657Z [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
2022-04-13T21:54:31.3842167Z [ERROR] Re-run Maven using the -X switch to enable full debug logging.
2022-04-13T21:54:31.3842497Z [ERROR] 
2022-04-13T21:54:31.3842828Z [ERROR] For more information about the errors and possible solutions, please read the following articles:
2022-04-13T21:54:31.3843270Z [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
2022-04-13T21:54:31.4117891Z ##[error]Process completed with exit code 1.
```

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -21,7 +21,7 @@ jobs:
         server-password: MAVEN_CENTRAL_TOKEN
         gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }}
         gpg-passphrase: MAVEN_GPG_PASSPHRASE
-    - run: mvn deploy
+    - run: mvn clean deploy
       env:
         MAVEN_USERNAME: ${{ secrets.MAVEN_USERNAME }}
         MAVEN_CENTRAL_TOKEN: ${{ secrets.MAVEN_CENTRAL_TOKEN }}

